core/drivers/crypto/caam/hal/common/hal_cfg_dt.c

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2017-2019 NXP
+ *
+ * Brief   CAAM Configuration.
+ */
+#include <caam_common.h>
+#include <caam_hal_cfg.h>
+#include <caam_hal_jr.h>
+#include <caam_jr.h>
+#include <kernel/dt.h>
+#include <kernel/generic_boot.h>
+#include <libfdt.h>
+
+static const char *dt_ctrl_match_table = {
+	"fsl,sec-v4.0-ctrl",
```

Comment:
This node is not found in any upstream device tree, breaking boot on any board with a mainline kernel and dt:
```
I/TC: OP-TEE version: 3.6.0-141-ga41d3b18 (gcc version 8.2.1 20181130 (OSELAS.Toolchain-2018.12.0 8-20181130)) #1 Fri Sep  6 07:31:41 UTC 2019 arm
D/TC:0 0 check_ta_store:638 TA store: "early TA"
D/TC:0 0 check_ta_store:638 TA store: "Secure Storage TA"
D/TC:0 0 check_ta_store:638 TA store: "REE [buffered]"
D/TC:0 0 early_ta_init:255 Early TA bc50d971-d4c9-42c4-82cb-343fb7f37896 size 632165 (compressed, uncompressed 1037980)
D/TC:0 0 mobj_mapped_shm_init:447 Shared memory address range: 4e400000, 50400000
E/TC:0 0 plat_rng_init:354 Warning: seeding RNG with zeroes
D/TC:0 0 imx_wdog_base:142 path: /soc/aips-bus@2000000/wdog@20bc000
E/TC:0 0 caam_hal_cfg_get_conf:79 Caam Node not found err = 0xFFFFFFFF
E/TC:0 0 crypto_driver_init:79 CAAM Driver initialization (0xffff000a)
E/TC:0 0 Panic at core/drivers/crypto/caam/caam_ctrl.c:80 <crypto_driver_init>
E/TC:0 0 Call stack:
E/TC:0 0  0x4e0056fd print_kernel_stack at /ptx/work/WORK_BEEMI/rcz/backup/git/trees/optee/imx6q/core/arch/arm/kernel/unwind_arm32.c:450
```

Reply:
@Emantor,

You can find the linux version in [Code Aurora](https://wiki.codeaurora.org/xwiki/bin/i.MX+Project/)

Reply:
My expectation was that an upstream OP-TEE works with an upstream kernel and does not require undocumented device-tree properties not documented anywhere.

Reply:
ok, I have to think how to deal with. I'll submit a new commit (thru a new PR) for that.

Reply:
IMO this is not the way forward here. Working platforms should still work after the PR is merged, so the necessary change should be included with the commit which introduces the dependency on the device tree property.

Reply:
I tend to agree with @Emantor on this, we should be very careful with DT bindings and specs, otherwise things will become unmanageable (I mean difficult to use and possibly inconsistent).

ff you could get the DT stuff and linux drivers to be accepted into upstream Linux at about the same time we accept the secure drivers here, that would be great.

Reminds me of how we dealt with the secure console binding ([/secure-chosen/stdout-path](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/arm/secure.txt)). I first created a PR here, then proposed the binding to Linux. Rob H. made some suggestions and accepted the binding as long as I showed it was about to be used by OP-TEE and we were done.